Running e2e tests on different versions of vanilla k8s

We will use https://github.com/kubernetes-sigs/kind to test ECK against different versions of vanilla k8s.

If we will go with `kind`:
- [x] Setup new build agent on CI, approx `n1-standard-16` with 16 CPU and 60 GB RAM should be enough (details for setup: https://github.com/elastic/infra/blob/master/terraform/providers/gcp/env/elastic-apps/helm/values/gobld/devops-ci.yml)
- [x] Build new image or simply download `kind` binary to existed one?
- [x] Create new CI job (both JJBB yml and Jenkins pipeline)

If we will go with Ansible:
- Evaluate possibility of using Ansible playbook for bootstrapping K8s cluster
- Build Docker image to run Ansible
- How to handle SSH keys?
- Create new CI job (both JJBB yml and Jenkins pipeline)

Here is the error when the job fails:
```
Step 9/16 : RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 		go build             -mod readonly 			-ldflags "$GO_LDFLAGS" -tags="$GO_TAGS" -a 			-o elastic-operator github.com/elastic/cloud-on-k8s/cmd
---> Running in fbaada888f39

# github.com/elastic/cloud-on-k8s/pkg/controller/common/license
pkg/controller/common/license/check.go:35:22: undefined: publicKeyBytes

The command '/bin/sh -c CGO_ENABLED=0 GOOS=linux GOARCH=amd64 		go build             -mod readonly 			-ldflags "$GO_LDFLAGS" -tags="$GO_TAGS" -a 			-o elastic-operator github.com/elastic/cloud-on-k8s/cmd' returned a non-zero code: 2
Makefile:319: recipe for target 'docker-build' failed
```

`build/ci/e2e/vanilla_k8s.jenkinsfile` calls the `kind-e2e` make target which calls `set-kind-e2e-image` which calls `docker-build`. 

During the Docker build, the go binary build fails because the variable `publicKeyBytes` is not defined. 

This variable is not defined because `go generate -tags='$(GO_TAGS)' ./pkg/... ./cmd/...` has not yet been executed.

We can add (duplicate) the `go generate` command directly in the Dockerfile or call the `generate-crds` target in the `set-kind-e2e-image` target. Not sure what is the best. First option seems to me a little better as this `go generate` is part of the build at a same level as `go mod` or `go build` and should never been forgotten.

Last job failure is due to a compilation error: `publicKeyBytes redeclared in this block`.
```sh
# github.com/elastic/cloud-on-k8s/pkg/controller/common/license
21:18:18  pkg/controller/common/license/zz_generated.pubkey.go:7:5: publicKeyBytes redeclared in this block
21:18:18  	previous declaration at pkg/controller/common/license/pubkey_dev.go:9:5
21:18:40  FAIL	github.com/elastic/cloud-on-k8s/test/e2e [build failed]
```

Reproduced it in local, this is the output of the job that runs:
* `E2E_JSON=json test/e2e/run.sh -run ^Test -args -testContextPath ./testcontext.json`
* => `go test -v -failfast -timeout=4h -tags=e2e -p=1 github.com/elastic/cloud-on-k8s/test/e2e -run '^Test' -args -testContextPath ./testcontext.json`

The compilation failed because we did not [clean](https://github.com/elastic/cloud-on-k8s/blob/214060e3420a356c00b6e5e0d59f62e90ec5c88a/Makefile#L119-L120) (`rm .../zz_generated.pubkey.go`) before executing [`e2e-run`](https://github.com/elastic/cloud-on-k8s/blob/214060e3420a356c00b6e5e0d59f62e90ec5c88a/Makefile#L491) like this is done in the [`e2e`](https://github.com/elastic/cloud-on-k8s/blob/214060e3420a356c00b6e5e0d59f62e90ec5c88a/Makefile#L359-L360) target.

--- a/Makefile
+++ b/Makefile
@@ -488,7 +488,7 @@ kind-e2e: kind-node-variable-check set-kind-e2e-image e2e-docker-build
 	./hack/kind/kind.sh \
 		--load-images $(OPERATOR_IMAGE),$(E2E_IMG) \
 		--nodes 3 \
-		make e2e-run OPERATOR_IMAGE=$(OPERATOR_IMAGE)
+		make clean e2e-run OPERATOR_IMAGE=$(OPERATOR_IMAGE)
 
 ## Cleanup
 delete-kind:

